addons/hr_recruitment_survey/models/survey_invite.py

```diff
@@ -1,22 +1,25 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import fields, models, _
+from odoo.tools.misc import clean_context
 
 
 class SurveyInvite(models.TransientModel):
     _inherit = "survey.invite"
 
-    applicant_id = fields.Many2one('hr.applicant', string='Applicant', default=lambda self: self.env.context.get('active_id', None))
+    applicant_id = fields.Many2one('hr.applicant', string='Applicant')
 
     def action_invite(self):
         self.ensure_one()
 
-        if not self.applicant_id.response_id:
-            response = self.applicant_id.survey_id._create_answer(partner=self.applicant_id.partner_id)
-            self.applicant_id.response_id = response.id
-
-        body = _('The survey has been sent to "%s".', self.applicant_id.partner_name)
-        self.applicant_id.message_post(body=body)
+        if self.applicant_id:
+            if not self.applicant_id.response_id:
+                survey = self.applicant_id.survey_id.with_context(clean_context(self.env.context)) 
```

Comment:
Would use ``self.survey_id`` as this is the survey on which we are sending the invite. Applicant survey should be synchronized but the main one is the invite defined one.